caffe2/serialize/inline_container.h

```diff
@@ -119,135 +134,202 @@ class PyTorchStreamReader {
-                              " be corrupted or is not actually a PyTorch file.");
-   }
-   uint64_t file_format_version = read64BitIntegerLittleEndian();
-   if (file_format_version > kMaxSupportedFileFormatVersion) {
-     std::ostringstream errmsg;
-     errmsg << "Attempted to read a PyTorch file with version " << file_format_version
-            << " but the maximum supported version for reading is " << kMaxSupportedFileFormatVersion
-            << ". Your PyTorch installation may be too old.";
-     throw std::runtime_error(errmsg.str());
-   }
-   seekToNextAlignmentBoundary();
-  }
-  void readAndValidateFileFooter() {
+  bool readAndValidateFileHeader() {
+    // Validate magic number
+    cursor_ = 0;
+    in_.seekg(cursor_);
+    uint64_t magic = read64BitIntegerLittleEndian();
+    if (magic != kFileMagicNumber) {
+      throw std::runtime_error(
+          "Magic number mismatch in PyTorch file. File may"
+          " be corrupted or is not actually a PyTorch file.");
+    }
+    uint64_t file_format_version = read64BitIntegerLittleEndian();
+    if (file_format_version > kMaxSupportedFileFormatVersion) {
+      std::ostringstream errmsg;
+      errmsg << "Attempted to read a PyTorch file with version "
+             << file_format_version
+             << " but the maximum supported version for reading is "
+             << kMaxSupportedFileFormatVersion
+             << ". Your PyTorch installation may be too old.";
+      throw std::runtime_error(errmsg.str());
+    }
+    seekToNextAlignmentBoundary();
+    return true;
+  }
+
+  bool readAndValidateFileFooter() {
     // Seek to location of file footer. We've already validated that the file
     // length is a multiple of the alignment size
-    cursor = file_size - kFieldAlignment;
-    in.seekg(cursor);
+    cursor_ = file_size_ - kFieldAlignment;
+    in_.seekg(cursor_);
     auto tag = read64BitIntegerLittleEndian();
     if (tag != RecordTags::FOOTER) {
       throw std::runtime_error("File footer has wrong record type. Is this"
                                " file corrupted?");
     }
-    last_record_offset = read64BitIntegerLittleEndian();
-    if (last_record_offset > file_size) {
+    last_record_offset_ = read64BitIntegerLittleEndian();
+    if (last_record_offset_ > file_size_) {
       throw std::runtime_error("Offset of last record is higher than the size"
                                " of the file! Is this file corrupted?");
     }
+    return true;
   }
 };
 
-class PyTorchStreamWriter {
+class PyTorchStreamWriter final {
  public:
-  PyTorchStreamWriter(std::ostream& out_) : out(out_) {
+  PyTorchStreamWriter(std::ostream& out) : out_(out) {
     writeFileHeader();
     // In the case that we do not write any records into this file, the last
     // record index written into the footer will point to the footer itself.
-    last_record_idx = cursor;
+    last_record_idx_ = cursor_;
   }
+
   uint64_t writeRecord(const char* data, size_t size) {
```

Comment:
I'd suggest to make it something like

std::uint64_t WriteRecord(const void* data, const std::size_t size) { ... }